File: src/main/kotlin/com/invasionofsmallcubes/traveldodolist/Trip.kt
package com.invasionofsmallcubes.traveldodolist

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "trips")
data class Trip(@Id var id:String? = null, val departureAirport: String,
                val arrivalAirport: String, val departureDate: String,
                val arrivalDate: String, var owner: String)

File: src/main/javascript/src/Trip.ts
import {Moment} from "moment";

export default class Trip {
    public readonly arrivalDate: Moment;
    public readonly departureDate: Moment;
    public readonly arrivalAirport: string;
    public readonly departureAirport: string;
    public readonly id: string;

    constructor(id: string, arrivalDate: Moment, departureDate: Moment, arrivalAirport: string, departureAirport: string) {
        this.arrivalAirport = arrivalAirport;
        this.arrivalDate = arrivalDate;
        this.departureAirport = departureAirport;
        this.departureDate = departureDate;
        this.id = id;
    }
}